pkg/agent/cniserver/pod_configuration_windows.go

```diff
@@ -46,6 +46,8 @@ func (pc *podConfigurator) connectInterfaceToOVSAsync(ifConfig *interfacestore.I
 	go func() {
 		klog.Infof("Waiting for interface %s to be created", hostIfAlias)
 		err := wait.PollImmediate(time.Second, 60*time.Second, func() (bool, error) {
+			containerAccess.lockContainer(containerID)
```

Comment:
What does it protect from? If it's subsequent CNI calls, won't containerID be different from the first one? or it's different in containerd? we use `getInfraContainer` to get the lock identity in CNIAdd.
And this reminds me what if the first CNI call and the subquent ones run into connectInterfaceToOVSAsync, will duplicate ports being created or it has been handled?

Reply:
@tnqn The lock is to protect multiple CNI all on same container/Pod, and to keep OVS port creation logic in one transaction.

>And this reminds me what if the first CNI call and the subquent ones run into connectInterfaceToOVSAsync, will duplicate ports being created or it has been handled?

Yes, you're right. `connectInterfaceToOVSAsync ` has chanced to be called more that once. But only one port will be created because the following port creation attemp will fail due to "duplicate OVS port error" from OVSDB.

I think we could check if OVS port has been created in `connectInterfaceToOVSAsync ` as enhancement to avoid "dup OVS port creration error":
https://github.com/vmware-tanzu/antrea/blob/main/pkg/agent/cniserver/pod_configuration_windows.go#L62

What do you think?

Reply:
I got answer for the second question and the solution sounds good to me.

For the first question, I think using containerID to get the lock cannot really prevent the race condition as the containerID will be different in multiple CNI calls: https://github.com/vmware-tanzu/antrea/blob/main/pkg/agent/cniserver/server_windows.go#L78, unless the havior in containerd is different from docker.

Reply:
@tnqn ContainerD has different behavior compared with Docker. ContainerD only send CNI request for sanbdox just like on Linux.

On the other hand,  antrea only tries to create OVS port for infra container with following flag:
https://github.com/vmware-tanzu/antrea/blob/main/pkg/agent/cniserver/pod_configuration.go#L215